django_refugeerights/subscription/views.py

```diff
@@ -13,3 +17,27 @@ class SubscriptionViewSet(viewsets.ModelViewSet):
     queryset = Subscription.objects.all()
     serializer_class = SubscriptionSerializer
     filter_fields = ('contact_key', 'to_addr', 'active', )
+
+
+class SwitchSubscriptionView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, format=None):
+        data = request.data
+        try:
+            contact_key = data['contact_key']
+        except KeyError:
+            raise ValidationError("A contact 'key' parameter is required.")
+        try:
+            messageset_id = data['messageset_id']
+        except KeyError:
+            raise ValidationError("A messageset 'id' parameter is required.")
+        return HttpResponse(status=200)
+        try:
+            messageset = MessageSet.objects.filter(pk=messageset_id).exists()
+        except KeyError:
+            raise ValidationError("A messageset 'id' doesn't exist")
+        try:
+            subscription = Subscription.objects.filter(pk=contact_key).update(active=False)
```

Comment:
We also don't need the `try: ... except: ...` here -- the subscription update should always work.